File: homework/Nov23/miniProject/Page/ProductListPage.java
package lab.Nov26.miniProject.Page;

import lab.Nov26.miniProject.Product;
import lab.Nov26.miniProject.ProductManager;
import lab.Nov26.miniProject.State;
import lab.Nov26.miniProject.Abstract.Page;

public class ProductListPage extends Page {
	
	public ProductListPage() {}

	@Override
	public void show() {
		ProductManager products = State.getState().getProductManager();
		products.showProducts();
		System.out.println("Type product Id to buy\nb. back\nc. checkout");
		String input = State.getState().getScanner().next();
		
		switch (input) {
			case "b": {
				Page previousPage = (Page) State.getState().getStack().pop();
				previousPage.show();
				break;
			}
			case "c": {
				State.getState().getStack().push(this);
				new CheckoutPage().show();
				break;
			}
			default: {
				if(isNumber(input)) {
					int id = Integer.parseInt(input);
					Product p = products.getProduct(id);
					if(products.isContain(p)) {
						State.getState().getShoppingCart().addProduct(p);
					}
					else {
						System.out.println("Id not found, Please try again");
					}
				}
				show();
				break;
			}
		}
	}
	
}

File: src/lab/miniProject_TS/src/page/productListPage.ts
import { Page } from "../interface"
import Product from "../product"
import ProductManager from "../productManager"
import State from "../state"
import CheckoutPage from "./checkoutPage"

class ProductListPage implements Page {
    constructor() {}
    public show(): void {
        const products: ProductManager = State.getState().getProductManager()
        products.showProducts()
        const rl = State.getState().getRl()
        rl.question('Type product Id to buy\nb. back\nc. checkout\n', (input: string) => {
            switch (input) {
                case 'b':
                    State.getState().getStack().pop()?.show()
                    break
                case 'c': 
                    State.getState().getStack().push(this)
                    new CheckoutPage().show()
                    break
                default:
                    if(!isNaN(+input)) {
                        let id: number = +input
                        let p: Product = products.getProduct(id)
                        if (products.isContain(p)) State.getState().getShoppingCart().addProduct(p)
                        else console.log('Id not found, Please try again')
                    }
                    this.show()
                    break
            }
        })
    }

}

export default ProductListPage
